tools/genjava.js

```diff
@@ -181,9 +181,10 @@ function generateClass(cls) {
   if ( foam.Array.isInstance(cls) ) {
     cls = cls[1];
   }
-  if ( typeof cls === 'string' )
+  if ( typeof cls === 'string' ) {
     cls = foam.lookup(cls);
-
+    cls = cls.buildJavaClass();
```

Comment:
I think this is going to break refining java classes because the will force the java to be generated before it has a chance to be refined.

I think we'll need to call this again if we load a refinement, which will be difficult if it's an interface or ancestor. Why did we make this change again? So that the java definitions would be available earlier for sub-classes? Maybe we should put functions that generate the class on the fly in the CLASSES map instead.

FYI: @adamfox-nanopay

Reply:
Not a problem with this PR, but it will be a problem for Adam.